packages/x-date-pickers-pro/src/DateRangePicker/DateRangePickerView.tsx

```diff
@@ -143,10 +141,10 @@ function DateRangePickerViewRaw<TInputDate, TDate>(
     shouldDisableDate &&
     ((dayToTest: TDate) => shouldDisableDate?.(dayToTest, currentlySelectingRangeEnd));
 
-  const [start, end] = parsedValue;
+  const [start, end] = value;
   const { changeMonth, calendarState, onMonthSwitchingAnimationEnd, changeFocusedDay } =
-    useCalendarState({
-      value: start || end,
+    useCalendarState<TDate>({
+      value: start ?? end,
```

Comment:
This would change the behavior if  `start` is `null`?

Reply:
> This would change the behavior if start is null?

No, `null` also triggers the fallback here
The diff would be if `start === 0` or `start === ''`
But I agree it's not purely related to this PR.
I tend to remove the `date || xxx` because if we have a date library storing dates as timestamp it can blow in our face.
With that being said, that's probably very unlikely

Removed